mudslide/models.py

```diff
@@ -553,12 +562,205 @@ def dV(self, X: ArrayLike) -> ArrayLike:
+        self.states = states
+        self.nstates_ = len(self.states)
+        self.sub_dir_stem = sub_dir_stem
+        self.sub_dir_num = 0
+
+        self.turbomole_init()
+
+    def turbomole_init(self):
+        self.coord_path = Path(self.turbomole_dir)/"control"  
+        self.get_coords()
+        self.call_and_parse_turbomole()
+
+    def get_coords(self):
+        with open(self.coord_path, "r") as coord_file:
+            lines = coord_file.readlines()
+
+        coords = []
+        self.atom_order = []
+
+        try:
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+        except:
+            
+            self.coord_path = Path(self.turbomole_dir)/"coord"
+            with open(self.coord_path, "r") as coord_file:
+                lines = coord_file.readlines()
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+
+        first_char = ""
+        current_line = read_start + 1
+
+        while True:
+            line = lines[current_line]
+            first_char = line[0]
+            if first_char == "$":
+                break
+            line_split = re.split("\s+", line.strip())
+            self.atom_order.append(line_split[-1])
+            coords.extend([float(val) for val in line_split[0:3]])
+            current_line += 1
+    
+        self.ndim_ = len(coords)
+        self.X = np.array(coords, dtype=np.float64).reshape(self.ndim())
+        self.mass = [masses[atom] for atom in self.atom_order]
+        self.mass = [3 * [masses[atom]] for atom in self.atom_order]
+        self.mass = np.array(self.mass, dtype=np.float64).reshape(self.ndim()) * amu_to_au
+
+
+    def update_coords(self, X):
+
+        first_space = 4 * " "
+        first_space_neg = 3 * " "
+
+        space = 6 * " "
+        space_neg = 5 * " "
+
+        X = X.reshape((self.ndim() // 3, 3))
+
+        with open(self.coord_path, "r") as coord_file:
+            lines = coord_file.readlines()
+
+
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+
+        with open(self.coord_path, "w") as coord_file:
+            current_line = read_start + 1
+
+            for i, coord in enumerate(X):  
+                if coord[0] < 0:
+                    s1 = first_space_neg
+                else:
+                    s1 = first_space
+
+                if coord[1] < 0:
+                    s2 = space_neg
+                else:
+                    s2 = space
+
+                if coord[2] < 0:
+                    s3 = space_neg
+                else:
+                    s3 = space
+                lines[current_line] = s1 + f"{coord[0]:1.14f}" + s2 + f"{coord[1]:1.14f}" + s3 + f"{coord[2]:1.14f}" + space + self.atom_order[i] + "\n"  
+                current_line+=1
+            coord_file.write("".join(lines))            
+
+
+    def call_and_parse_turbomole(self, outname="turbo.out"):
```

Comment:
This needs a more descriptive name. For one, the parse part can be implied. But also, if you're giving it such a general name (`call`), then it should be flexible enough to handle just about any call you'd want to make. Right now it's hardcoded for `ridft`, `rdgrad`, and `egrad`. What if you wanted to customize that?

Reply:
If you could provide some details on what exactly we want to achieve - would be great! Thanks!

Reply:
Hopefully this is now addressed! now 
<img width="746" alt="Screen Shot 2022-04-10 at 2 36 43 PM" src="https://user-images.githubusercontent.com/46786940/162634443-0710578d-60be-428b-95d0-5066a17b2818.png">

<img width="748" alt="Screen Shot 2022-04-10 at 2 37 43 PM" src="https://user-images.githubusercontent.com/46786940/162634489-1ac7eb27-bc2b-4bf5-8bd1-10c4796504c6.png">